src/link_prober/LinkProberStateMachineBase.cpp

```diff
@@ -100,10 +119,20 @@ void LinkProberStateMachineBase::processEvent(T &t)
 {
     LinkProberState *currentLinkProberState = dynamic_cast<LinkProberState *> (getCurrentState());
     LinkProberState *nextLinkProberState = currentLinkProberState->handleEvent(t);
-    if (nextLinkProberState != currentLinkProberState) {
-        postLinkManagerEvent(nextLinkProberState);
+    if (nextLinkProberState == nullptr) {
+        MUXLOGERROR(
+            boost::format(
+                "%s: link prober state %d could not handle event"
+            ) %
+            mMuxPortConfig.getPortName() %
+            currentLinkProberState->getStateLabel()
+        );
+    } else {
```

Comment:
Can you explain under what circumstances it will enter this block of code?

Reply:
In `LinkProberState`, I've changed the following event handler method functions to virtual functions and added a default implementation that will return a `nullptr`.
```cpp
LinkProberState::handleEvent(IcmpPeerEvent &event)
LinkProberState::handleEvent(IcmpSelfEvent &event)
LinkProberState::handleEvent(IcmpUnknownEvent &event)
LinkProberState::handleEvent(IcmpPeerActiveEvent &event)
LinkProberState::handleEvent(IcmpPeerUnknownEvent &event)
```
Actually, I believe this code block will never hit the condition: `nextLinkProberState == nullptr`, because we always assume self state only process self probe event and peer state only process peer probe event. This condition only acts as a failover check.